### PR TITLE
Fix bug that study level can be null and include more refactoring of person to student

### PR DESCRIPTION
**Pull request by enhao25**

### CODE REVIEW

**Review by JonahhGohh — APPROVED** [review]
LGTM! 😄

File: docs/tutorials/AddRemark.md [hunk 1]
```diff
@@ -222,7 +222,7 @@ If you are stuck, check out the sample
 
 ## Add `Remark` to the model
 
-Now that we have all the information that we need, let’s lay the groundwork for propagating the remarks added into the in-memory storage of student data. We achieve that by working with the `Person` model. Each field in a Person is implemented as a separate class (e.g. a `Name` object represents the student’s name). That means we should add a `Remark` class so that we can use a `Remark` object to represent a remark given to a student.
+Now that we have all the information that we need, let’s lay the groundwork for propagating the remarks added into the in-memory storage of student data. We achieve that by working with the `Student` model. Each field in a Student is implemented as a separate class (e.g. a `Name` object represents the student’s name). That means we should add a `Remark` class so that we can use a `Remark` object to represent a remark given to a student.
```
Comment (nowknowing) [inline docs/tutorials/AddRemark.md]: Do we keep this? Remarks isn't in our milestone but I suppose we could

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 2]
```diff
@@ -47,18 +47,18 @@ public void parseCommand_clear() throws Exception {
 
     @Test
     public void parseCommand_delete() throws Exception {
```
Comment (nowknowing) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java]: Should this also be parseStudentCommand?

File: src/test/java/seedu/address/testutil/StudentUtil.java [hunk 3]
```diff
@@ -21,13 +21,13 @@
      * Returns an add command string for adding the {@code student}.
      */
     public static String getAddCommand(Student student) {
```
Comment (nowknowing) [inline src/test/java/seedu/address/testutil/StudentUtil.java]: getAddStudentCommand

**Review by nowknowing — CHANGES_REQUESTED** [review]
I'm not sure about keeping AddRemark, and having to add 'student' to some of the parsing commands.

File: docs/tutorials/AddRemark.md (hunk 1 above)
Comment (JonahhGohh) [reply docs/tutorials/AddRemark.md]: We do not actually need this file. It is from the tutorial for tP from week 6. I don't even know why this is here. But I guess the change is harmless.

Comment (enhao25) [reply docs/tutorials/AddRemark.md]: Ok ah I will change the values in the file back to the original code, I have no idea why this file is in our directory also.

Comment (enhao25) [reply docs/tutorials/AddRemark.md]: Updated.

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java (hunk 2 above)
Comment (enhao25) [reply src/test/java/seedu/address/logic/parser/AddressBookParserTest.java]: Updated.

File: src/test/java/seedu/address/testutil/StudentUtil.java (hunk 3 above)
Comment (enhao25) [reply src/test/java/seedu/address/testutil/StudentUtil.java]: Updated.

**Review by nowknowing — APPROVED** [review]

**Pull request merged**